sage-ios/SAGE/SAGE/SelectAnnouncementSchoolTableViewController.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  SelectAnnouncementSchoolTableViewController.swift
+//  SAGE
+//
+//  Created by Sameera Vemulapalli on 11/24/15.
+//  Copyright © 2015 Cal Blueprint. All rights reserved.
+//
+
+import UIKit
+
+class SelectAnnouncementSchoolTableViewController: UITableViewController {
+    
+    var schools: [School]?
+    var activityIndicator: UIActivityIndicatorView = UIActivityIndicatorView(activityIndicatorStyle: .White)
+    weak var parentVC: AddAnnouncementController?
+    var currentErrorMessage: ErrorView?
+    let everyoneSchool = School(id: -1, name: "Everyone", students: [User]())
+    
+    init() {
+        super.init(style: .Plain)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = "Choose School"
+        self.tableView.tableFooterView = UIView()
+        
+        self.view.addSubview(self.activityIndicator)
+        self.activityIndicator.startAnimating()
+        
+        self.refreshControl = UIRefreshControl()
+        self.refreshControl?.backgroundColor = UIColor.mainColor
+        self.refreshControl?.tintColor = UIColor.whiteColor()
+        self.refreshControl?.addTarget(self, action: "loadSchools", forControlEvents: .ValueChanged)
+        
+        self.loadSchools()
+    }
+    
+    func loadSchools() {
+        AdminOperations.loadSchools({ (schoolArray) -> Void in
+            self.schools = schoolArray
+            self.schools?.insert(self.everyoneSchool, atIndex: 0)
+            self.tableView.reloadData()
+            self.activityIndicator.stopAnimating()
+            self.refreshControl?.endRefreshing()
+            }) { (errorMessage) -> Void in
+                self.showErrorAndSetMessage(errorMessage)
+        }
+    }
+    
+    func showErrorAndSetMessage(message: String) {
+        let error = self.currentErrorMessage
+        let errorView = super.showError(message, currentError: error, color: UIColor.mainColor)
+        self.currentErrorMessage = errorView
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        self.activityIndicator.centerHorizontally()
+        self.activityIndicator.centerVertically()
+    }
+    
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if let schools = self.schools {
+            return schools.count + 1
+        } else {
+            return 0
+        }
+    }
+    
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let school = self.schools![indexPath.row]
```

Comment:
fixed the row offset! I was planning on handling when the request was actually made - in the meantime, the school object that handles everyone is passed back and the label in the view can be set to "everyone" as selected